soh/src/code/z_map_exp.c

```diff
@@ -855,58 +858,50 @@ void Minimap_Draw(PlayState* play) {
 
                     gDPSetPrimColor(OVERLAY_DISP++, 0, 0, minimapColor.r, minimapColor.g, minimapColor.b, interfaceCtx->minimapAlpha);
 
+                    s16 iconSize = 8;
+
                     if (((play->sceneNum != SCENE_SPOT01) && (play->sceneNum != SCENE_SPOT04) &&
                          (play->sceneNum != SCENE_SPOT08)) ||
                         (LINK_AGE_IN_YEARS != YEARS_ADULT)) {
-                        bool Map0 = gMapData->owEntranceIconPosY[sEntranceIconMapIndex] << 2 == 0;
-                        s16 IconSize = 8;
-                        s16 PosX = gMapData->owEntranceIconPosX[sEntranceIconMapIndex] + (Map0 ? 0 : X_Margins_Minimap);
-                        s16 PosY = gMapData->owEntranceIconPosY[sEntranceIconMapIndex] + (Map0 ? 0 : Y_Margins_Minimap);
-                        //gFixDungeonMinimapIcon fix both Y position of visible icon and hide these non needed.
-
-                        s16 TopLeftX = (Map0 ? OTRGetRectDimensionFromLeftEdge(PosX) : OTRGetRectDimensionFromRightEdge(PosX)) << 2;
-                        s16 TopLeftY = PosY << 2;
-                        s16 TopLeftW = (Map0 ? OTRGetRectDimensionFromLeftEdge(PosX + IconSize) : OTRGetRectDimensionFromRightEdge(PosX + IconSize)) << 2;
-                        s16 TopLeftH = (PosY + IconSize) << 2;
+                        u8 topLeft0 = gMapData->owEntranceIconPosY[sEntranceIconMapIndex] == 0;
+                        // The game authentically uses larger negative values for the entrance icon Y pos value. Normally only the first 12 bits
+                        // would be read when the final value is passed into `gSPTextureRectangle`, but our cosmetic hud placements requires using
+                        // `gSPWideTextureRectangle` which reads the first 24 bits instead. This caused the icon to be placed off screen.
+                        // To address this, we take only the first 10 bits (which are later left-shifted by 2 to get our final 12 bits)
+                        // to fix the entrance icon position when used with `gSPWideTextureRectangle`
+                        s16 newY = gMapData->owEntranceIconPosY[sEntranceIconMapIndex] & 0x3FF;
+                        s16 PosX = gMapData->owEntranceIconPosX[sEntranceIconMapIndex] + (topLeft0 ? 0 : X_Margins_Minimap);
+
+                        s16 entranceX = topLeft0 ? OTRGetRectDimensionFromLeftEdge(PosX) : OTRGetRectDimensionFromRightEdge(PosX);
```

Comment:
a comment explaining how the `topLeft0` logic is working would be nice

it's not clear why `entranceX` sometimes should get set from the left and other times from the right